.github/workflows/check.yml

```diff
@@ -1,43 +1,87 @@
 name: Check
 
-# XXX disabled for now because we use Jenkins still, but at the time this was tested it was functional
-# on:
-#   pull_request:
-#     branches: [ master ]
+on:
+  pull_request:
+    branches:
+      - '*'  # Trigger on all branches for pull requests
+  push:
```

Comment:
As mentioned in another comment I would suggest to move this trigger and the related actions to another workflow.
If you push your changes to a specific branch in the main repository, you can test the workflow by adding the branch you are using to the list of branches.